src/hydra-queue-runner/build-remote.cc

```diff
@@ -271,7 +271,11 @@ void State::buildRemote(ref<Store> destStore,
         }
 
         /* Copy the input closure. */
-        if (!machine->isLocalhost()) {
+        if (machine->isLocalhost()) {
+            StorePathSet closure;
+            destStore->computeFSClosure(inputs, closure);
+            copyPaths(*destStore, *localStore, closure, NoRepair, NoCheckSigs, NoSubstitute);
+        } else {
```

Comment:
I think we should probably have some counter activity here, just like in the non-localhost case.